openff/models/tests/test_types.py

```diff
@@ -36,37 +38,46 @@ class Atom(DefaultModel):
             foo=2.0 * unit.nanometer,
             fuu=2.0 * openmm.unit.nanometer,
             bar="90.0 degree",
+            barint="90 degree",
             baz=0.4 * openmm.unit.nanometer,
             qux=openmm.unit.Quantity(np.float64(0.4), openmm.unit.nanometer),
+            quix=openmm.unit.Quantity(2, openmm.unit.nanometer),
             quux=1,
             fnord=4.2,
             fum="fum",
             zot=["zot", 1, 4.2],
             fred={"baz": 2},
         )
 
-        assert a.mass == 4 * unit.atomic_mass_constant
-        assert a.charge == 0 * unit.elementary_charge
+        assert a.mass == 4.0 * unit.atomic_mass_constant
+        assert a.charge == 0.0 * unit.elementary_charge
+        assert isinstance(a.charge.m, float)
         assert a.foo == 2.0 * unit.nanometer
         assert a.fuu == 2.0 * unit.nanometer
         assert a.bar == 90 * unit.degree
+        assert a.barint == 90.0 * unit.degree
+        assert isinstance(a.barint.m, float)
         assert a.baz == 0.4 * unit.nanometer
         assert a.qux == 0.4 * unit.nanometer
-        assert a.quux == 1
+        assert a.quix == 2.0 * unit.nanometer
+        assert isinstance(a.quix.m, float)
+        assert a.quux == 1.0
```

Comment:
```suggestion
        assert isinstance(a.quux.m, float)
```
This is defined as `int` in the model, no `FloatQuantity` involved, so I think it should not be converted? This is an existing oversight but it'd be nice to fix it here if I'm not mistaken.

I guess equality of raw numbers case doesn't require the two objects to be of the same type:
```
>>> 1 == 1.0
True
```

kinda makes sense and I should have known this already

Reply:
yeah this was me being overzealous, it's untouched as an int, but `1 == 1.0`